presto-analyzer/src/main/java/com/facebook/presto/sql/analyzer/Analysis.java

```diff
@@ -900,6 +901,11 @@ public Optional<QuerySpecification> getCurrentQuerySpecification()
         return currentQuerySpecification;
     }
 
+    public List<String> getInvokedFunctionNames()
+    {
+        return ImmutableList.copyOf(functionHandles.values().stream().map(functionHandle -> functionHandle.getName()).collect(toSet()));
```

Comment:
you can use` .map(FunctionHandle::getCatalogSchemaName)`

Reply:
I meant 
`return functionHandles.values().stream().map(FunctionHandle::getName).collect(toImmutableList());`
why do we need another copy again ?

Reply:
Ahh okay I see it. we are using set to de-duplicate and then making a list out of it